x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/ProfilingIndexTemplateRegistry.java

```diff
@@ -0,0 +1,224 @@
+        new IndexTemplateConfig(
+            "profiling-events",
+            "/org/elasticsearch/xpack/profiler/index-template/profiling-events.json",
+            INDEX_TEMPLATE_VERSION,
+            PROFILING_TEMPLATE_VERSION_VARIABLE
+        ),
+        new IndexTemplateConfig(
+            "profiling-metrics",
+            "/org/elasticsearch/xpack/profiler/index-template/profiling-metrics.json",
+            INDEX_TEMPLATE_VERSION,
+            PROFILING_TEMPLATE_VERSION_VARIABLE
+        ),
+        new IndexTemplateConfig(
+            "profiling-executables",
+            "/org/elasticsearch/xpack/profiler/index-template/profiling-executables.json",
+            INDEX_TEMPLATE_VERSION,
+            PROFILING_TEMPLATE_VERSION_VARIABLE
+        ),
+        new IndexTemplateConfig(
+            "profiling-stackframes",
+            "/org/elasticsearch/xpack/profiler/index-template/profiling-stackframes.json",
+            INDEX_TEMPLATE_VERSION,
+            PROFILING_TEMPLATE_VERSION_VARIABLE
+        ),
+        new IndexTemplateConfig(
+            "profiling-stacktraces",
+            "/org/elasticsearch/xpack/profiler/index-template/profiling-stacktraces.json",
+            INDEX_TEMPLATE_VERSION,
+            PROFILING_TEMPLATE_VERSION_VARIABLE
+        ),
+        // templates for regular indices
+        new IndexTemplateConfig(
+            ".profiling-ilm-lock",
+            "/org/elasticsearch/xpack/profiler/index-template/.profiling-ilm-lock.json",
+            INDEX_TEMPLATE_VERSION,
+            PROFILING_TEMPLATE_VERSION_VARIABLE
+        ),
+        new IndexTemplateConfig(
+            "profiling-returnpads-private",
+            "/org/elasticsearch/xpack/profiler/index-template/profiling-returnpads-private.json",
+            INDEX_TEMPLATE_VERSION,
+            PROFILING_TEMPLATE_VERSION_VARIABLE
+        ),
+        new IndexTemplateConfig(
+            "profiling-sq-executables",
+            "/org/elasticsearch/xpack/profiler/index-template/profiling-sq-executables.json",
+            INDEX_TEMPLATE_VERSION,
+            PROFILING_TEMPLATE_VERSION_VARIABLE
+        ),
+        new IndexTemplateConfig(
+            "profiling-sq-leafframes",
+            "/org/elasticsearch/xpack/profiler/index-template/profiling-sq-leafframes.json",
+            INDEX_TEMPLATE_VERSION,
+            PROFILING_TEMPLATE_VERSION_VARIABLE
+        ),
+        new IndexTemplateConfig(
+            "profiling-symbols",
+            "/org/elasticsearch/xpack/profiler/index-template/profiling-symbols.json",
+            INDEX_TEMPLATE_VERSION,
+            PROFILING_TEMPLATE_VERSION_VARIABLE
+        )
+    );
+
+    @Override
+    protected Map<String, ComposableIndexTemplate> getComposableTemplateConfigs() {
+        return templatesEnabled ? COMPOSABLE_INDEX_TEMPLATE_CONFIGS : Collections.emptyMap();
+    }
+
+    public static boolean isAllTemplatesCreated(ClusterState state) {
```

Comment:
Nit: would you renaming to `areAllTemplatesCreated`?

Reply:
Done in https://github.com/elastic/elasticsearch/pull/95666/commits/f1dee21951b7bae18cd687618df69e5cd7987cd1.